include/deal.II/base/work_stream.h

```diff
@@ -663,6 +669,149 @@ namespace WorkStream
 #  endif // DEAL_II_WITH_TBB
 
 
+
+#  ifdef DEAL_II_WITH_TASKFLOW
+    /**
+     * Mostly a copy of the 2nd implementation of the Workstream paper taking
+     * advantage of thread local lists for re-use. Uses taskflow for task
+     * scheduling rather than TBB. Currently does not support chunking.
+     */
+
+
+    namespace taskflow_no_coloring
+    {
+      /**
+       * The main run function for the taskflow colorless implementation. The
+       * last two arguments in this function are for chunking support which
+       * currently does not exist but ideally will later. For now they are
+       * ignored but still here to permit existing programs to function
+       */
+      template <typename Worker,
+                typename Copier,
+                typename Iterator,
+                typename ScratchData,
+                typename CopyData>
+      void
+      run(const Iterator                             &begin,
+          const std_cxx20::type_identity_t<Iterator> &end,
+          Worker                                      worker,
+          Copier                                      copier,
+          const ScratchData                          &sample_scratch_data,
+          const CopyData                             &sample_copy_data,
+          const unsigned int /*queue_length*/ = 2 *
+                                                MultithreadInfo::n_threads(),
+          const unsigned int /*chunk_size*/ = 8)
+
+      {
+        tf::Executor &executor = MultithreadInfo::get_taskflow_executor();
+        tf::Taskflow  taskflow;
+
+        using ScratchDataList =
+          std::list<ScratchDataObject<Iterator, ScratchData, CopyData>>;
+
+        Threads::ThreadLocalStorage<ScratchDataList>
+          thread_safe_scratch_data_list;
+
+        tf::Task last_copier;
+
+        // idx is used to connect each worker to its copier as communication
+        // between tasks is not supported. It does this by providing a unique
+        // index in the vector of pointers copy_datas at which the copy data
+        // object where the work done by work task #idx is stored
+        unsigned int idx = 0;
+
+        std::vector<std::unique_ptr<CopyData>> copy_datas;
+
+        // Generate a static task graph. Here we generate a task for each cell
+        // that will be worked on. The tasks are not executed until all of them
+        // are created, this code runs sequentially.
+        for (Iterator i = begin; i != end; ++i, ++idx)
+          {
+            copy_datas.emplace_back();
+            // Create a worker task.
+            auto worker_task =
+              taskflow
+                .emplace([it = i,
+                          idx,
```

Comment:
This is not wrong, but it made me wonder whether there is a good reason to capture the iterator `i` via a local renamed variable `it` whereas you just capture `idx` with its old name. Doing the same thing in different ways is a good way to confuse the reader, so my preference would be to use one style or the other.

Reply:
agreed. @RyanMoulday Can you please rename ``i`` to ``it`` in the for loop above and then just capture ``it``?